### PR TITLE
CI: add tester for simplex tests

### PR DESCRIPTION
**Pull request by peterrum**
@tjhei Was this what you were thinking about?

### CODE REVIEW

**Comment by tjhei** [comment]
The OSX already takes about 60 minutes before your change, so this is not ideal.
Should we just add another linux variant to github actions?

**Comment by peterrum** [comment]
@tjhei How do you like the github action now? It is sad that we cannot test Trilinos and PETSc here (i.e. parallel applications); I guess we can only test it on CDast.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -29,3 +29,37 @@ jobs:
...
+    # simple parallel debug build using g++ with simplex configuration enabled
+
+    name: linux debug parallel simplex
+    runs-on: [ubuntu-18.04]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: modules
+      run: |
+        sudo apt-get install libopenmpi-dev numdiff
```
Comment (tjhei) [inline .github/workflows/linux.yml:43]:
```suggestion
        sudo apt-get install -y software-properties-common
        sudo add-apt-repository ppa:ginggs/deal.ii-9.2.0-backports
        sudo apt-get update
        sudo apt-get install -yq --no-install-recommends numdiff  libp4est-dev trilinos-all-dev petsc-dev
```

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -29,3 +29,37 @@ jobs:
...
+    - name: setup
+      run: |
+        ./contrib/utilities/download_clang_format
+    - name: info
+      run: |
+        mpicc -v
+        cmake --version
+    - name: configure
+      run: |
+        cmake -D CMAKE_BUILD_TYPE=Debug -D DEAL_II_CXX_FLAGS='-Werror' -D CMAKE_CXX_COMPILER="mpic++"  -D DEAL_II_WITH_MPI="ON" -D DEAL_II_WITH_SIMPLEX_SUPPORT="ON" -D DEAL_II_COMPONENT_EXAMPLES="OFF" .
```
Comment (tjhei) [inline .github/workflows/linux.yml:53]: do you need the CXX_COMPILER?

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -29,3 +29,37 @@ jobs:
...
+      uses: actions/upload-artifact@v1
+      with:
+        name: linux-simplex-detailed.log
+        path: detailed.log
+    - name: build
+      run: |
+        make -j 2
+    - name: test
+      run: |
+        make -j 2 setup_tests
```
Comment (tjhei) [inline .github/workflows/linux.yml:64]:
```suggestion
        make -j 2 setup_tests_simplex
```

File: .github/workflows/linux.yml [hunk 4]
```diff
@@ -29,3 +29,37 @@ jobs:
...
+        ctest -j 2 -R simplex/
```
Comment (tjhei) [inline .github/workflows/linux.yml:65]:
```suggestion
        ctest -j 2
```

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -29,3 +29,37 @@ jobs:
...
+    runs-on: [ubuntu-18.04]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: modules
+      run: |
+        sudo apt-get install libopenmpi-dev numdiff
+    - name: setup
+      run: |
+        ./contrib/utilities/download_clang_format
```
Comment (tjhei) [inline .github/workflows/linux.yml:46]: not needed?

File: .github/workflows/linux.yml (hunk 1 above)
Comment (peterrum) [reply .github/workflows/linux.yml]: Thanks @tjhei for loading also Trilinos and PETSc! I used the opportunity to enabled them during configuration! I also incorporated all your comments!

**Pull request merged**